cmd/software/deploy.go

```diff
@@ -104,8 +106,20 @@ func deployAirflow(cmd *cobra.Command, args []string) error {
 	if isDagOnlyDeploy {
 		return DagsOnlyDeploy(houstonClient, appConfig, ws, deploymentID, config.WorkingPath, nil, true)
 	}
+
+	// Fetch the description flag value from the command flags
+	desc, err := cmd.Flags().GetString("description")
+	if err != nil {
+		return err
+	}
+
+	// If the description is not set, use GetDefaultDeployDescription to get the default
```

Comment:
Why is the cli responsible for setting the default description? Shouldn't the API be handling this?

Reply:
agreed

Reply:
Perhaps it's because the API will reject a missing description?

Reply:
Do we not have a way to identify type of deploy on the backend? I am not familiar with houston code on this

Reply:
We can identify the type of deploy mechanism, and we can identify the client used.

Reply:
But, then even when the deployment is currently on dag-only, the command used to deploy could be `astro deploy` instead of `astro deploy --dags` so that's the grey area; knowing the exact command.